### ISSUE TITLE
stack dot --snapshot, --extra-deps

### ISSUE THREAD

**Issue by phadej** [commercialhaskell/stack#2214]
Currently `stack dot` has only `--external`, it would be cool if it could separate which packages come from  snapshot, which are extra-deps (from Hackage or git/local) and which are really local.

Related to: https://github.com/commercialhaskell/stack/issues/1265

Bonus: `stack list-dependencies` to take `--only-snapshot` to list only packages needed from snapshot.

I tried to look around, but I cannot find if the information is easily available somewhere

**Comment by mgsloan** [comment]
Makes sense!  Could even have the graph have some visual info of where the package version came from.
